nav2_rotation_shim_controller/src/nav2_rotation_shim_controller.cpp

```diff
@@ -207,10 +210,17 @@ geometry_msgs::msg::TwistStamped RotationShimController::computeVelocityCommands
 
     std::lock_guard<std::mutex> lock_reinit(mutex_);
     try {
-      geometry_msgs::msg::Pose sampled_pt_base = transformPoseToBaseFrame(getSampledPathPt());
-
-      double angular_distance_to_heading =
-        std::atan2(sampled_pt_base.position.y, sampled_pt_base.position.x);
+      auto sampled_pt = getSampledPathPt();
+      auto angular_distance_to_heading = [&](){
```

Comment:
This is a little odd as a lambda, can you just make this an if/else case to populate the `angular_distance_to_heading` value?